pgscatalog.utils/packages/pgscatalog.validate/src/pgscatalog/validate/cli/validate_cli.py

```diff
@@ -0,0 +1,226 @@
+    global data_sum
+    if os.path.exists(log_file):
+        log_result = _read_last_line(log_file)
+        if re.search("File is valid", log_result):
+            print("> valid\n")
+            data_sum['valid'].append(filename)
+        elif re.search("File is invalid", log_result):
+            print("#### invalid! ####\n")
+            data_sum['invalid'].append(filename)
+        else:
+            print("!! validation process had an issue. Please look at the logs.\n")
+            data_sum['other'].append(filename)
+    else:
+        print("!! validation process had an issue: the log file can't be found")
+        data_sum['other'].append(filename)
+
+
+def _check_args(args: argparse.Namespace) -> None:
+    global score_dir
+    ## Check parameters ##
+    # Type of validator
+    if args.type and args.type not in val_types:
+        print(f"Error: Validator type (option -t) '{args.t}' is not in the list of recognized types: {val_types}.")
+        exit(1)
+    # Logs dir
+    if args.log_dir and not os.path.isdir(args.log_dir):
+        print(f"Error: Log dir '{args.log_dir}' can't be found!")
+        exit(1)
+    # File and directory parameters (only one of the '-f' and '--dir' can be used)
+    if args.filename and args.dir:
+        print("Error: you can't use both options [-f] - single scoring file and [--dir] - directory of scoring files."
+              " Please use only 1 of these 2 options!")
+        exit(1)
+    elif not args.filename and not args.dir:
+        print("Error: you need to provide a scoring file [-f] or a directory of scoring files [--dir]!")
+        exit(1)
+    elif args.filename and not os.path.isfile(args.filename):
+        print(f"Error: Scoring file '{args.filename}' can't be found!")
+        exit(1)
+    elif args.dir and not os.path.isdir(args.dir):
+        print(f"Error: the scoring file directory '{args.dir}' can't be found!")
+        exit(1)
+    # Scoring files directory (only to compare with the harmonized files)
+    score_dir = None
+    if args.score_dir:
+        score_dir = args.score_dir
+        if not os.path.isdir(score_dir):
+            print(f"Error: Scoring file directory '{score_dir}' can't be found!")
+            exit(1)
+    elif args.type not in ['formatted', 'raw']:
+        print("WARNING: the parameter '--score_dir' is not present in the submitted command line,"
+              " therefore the comparison of the number of data rows between the formatted scoring file(s)"
+              " and the harmonized scoring file(s) won't be performed.")
+
+
+def _run_validator(filepath: str, log_dir: str, score_dir: str, check_filename: bool, header: bool) -> None:
+    """Run the file validator"""
+    # TODO: add check_filename and score_dir support
+    file = os.path.basename(filepath)
+    filename = file.split('.')[0]
+    print(f"# Filename: {file}")
+    if log_dir:
+        log_file = f'{log_dir}/{filename}_log.txt'
+    else:
+        log_file = None
```

Comment:
this all looks good but I personally prefer using [pathlib](https://docs.python.org/3/library/pathlib.html) when working with files in modern python (`os.path` is good too)